frontend/src/utils/is-on-tos-page.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Checks if the current page is the Terms of Service acceptance page.
+ * This function works outside of React Router context by checking window.location directly.
+ *
+ * @param {string} [pathname] - Optional pathname from React Router's useLocation hook
+ * @returns {boolean} True if the current page is the TOS acceptance page, false otherwise.
+ */
+export const isOnTosPage = (pathname?: string): boolean => {
```

Comment:
@amanape is there a better way to do this?

Approving for now because it's very much an improvement. But I don't know React well enough to judge this piece